napari/layers/points/_slice.py

```diff
@@ -0,0 +1,87 @@
+import warnings
+from dataclasses import dataclass, field
+from functools import cached_property
+from typing import Any, Optional, Tuple, Union
+
+import numpy as np
+
+from napari.layers.utils._slice_input import _SliceInput
+from napari.utils.transforms import Affine
+from napari.utils.translations import trans
+
+
+@dataclass(frozen=True)
+class _PointSliceResponse:
+    data: Any = field(repr=False)
+    indices: np.ndarray = field(repr=False)
+    scale: Any = field(repr=False)
+
+
+@dataclass(frozen=True)
+class _PointSliceRequest:
+    """Represents a single point slice request.
+    This should be treated a deeply immutable structure, even though some
+    fields can be modified in place.
+    In general, the execute method may take a long time to run, so you may
+    want to run it once on a worker thread.
+    """
+
+    dims: _SliceInput
+    data: Any = field(repr=False)
+    dims_indices: Any = field(repr=False)
+    data_to_world: Affine = field(repr=False)
+    size: Any = field(repr=False)
+    out_of_slice_display: bool = field(repr=False)
+
+    def execute(self) -> _PointSliceResponse:
+
+        slice_indices, scale = _PointSliceRequest._get_slice_data(
+            data=self.data,
+            ndim=self.dims.ndim,
+            dims_indices=self.dims_indices,
+            dims_not_displayed=self.dims.not_displayed,
+            size=self.size,
+            out_of_slice_display=self.out_of_slice_display,
+        )
+
+        return _PointSliceResponse(
+            data=self.data, indices=slice_indices, scale=scale
```

Comment:
For an image, the response's data is the sliced data. But here it's the same as the request's data. For the prototype, we had inheritance, so needed to put something in that field. But here, we don't.

I think we could consider removing this field or making it `self.data[slice_indices, :]` though that might be incur a performance penalty. Any thoughts? I haven't entirely thought this through yet.

Reply:
Excellent point. I would expect `self.data[slice_indices, :]` but I'll look further along to see what sort of performance impacts there could be by actually materializing the sliced data.

Reply:
Ah. I get it. Yeah, I'll remove it. We currently don't need access to that data, I don't see an immediate need, and it is easy to add if someone ends up needing it later on.